lib/thor/parser/option.rb

```diff
@@ -86,7 +86,7 @@ def usage(padding = 0)
       sample = "[#{sample}]" unless required?
 
       if boolean?
-        sample << ", [#{dasherize("no-" + human_name)}]" unless name == "force"
+        sample << ", [#{dasherize("no-" + human_name)}]" unless name == "force" or name.start_with?("no")
```

Comment:
I believe this line would prevent the automatic negation of all options that that start with `"no"`, even if they are not negative options?

For example, what if there was an boolean `--nougat` option. I believe this would prevent the generation of a `--no-nougat` option.